internal/controller/controller_test.go

```diff
@@ -596,6 +600,381 @@ var _ = Describe("Testing Route to HTTPProxy Controller", func() {
+			Expect(err).To(BeNil())
+
+			objRouteBar.Status = routev1.RouteStatus{
+				Ingress: []routev1.RouteIngress{
+					{
+						RouterName: RouterName,
+						Conditions: []routev1.RouteIngressCondition{
+							{
+								Status: v12.ConditionStatus(v1.ConditionTrue),
+								Type:   routev1.RouteAdmitted,
+							},
+						},
+					},
+				},
+			}
+			err = k8sClient.Status().Update(context.Background(), &objRouteBar)
+			Expect(err).To(BeNil())
+
+			rObjFoo := routev1.Route{}
+			err = k8sClient.Get(context.Background(), types.NamespacedName{Namespace: DefaultNamespace, Name: RouteFooName}, &rObjFoo)
+			Expect(err).To(BeNil())
+
+			rObjBar := routev1.Route{}
+			err = k8sClient.Get(context.Background(), types.NamespacedName{Namespace: DefaultNamespace, Name: RouteBarName}, &rObjBar)
+			Expect(err).To(BeNil())
+
+			time.Sleep(1 * time.Second)
+			httpProxyList := contourv1.HTTPProxyList{}
+			err = k8sClient.List(context.Background(), &httpProxyList, client.InNamespace(DefaultNamespace))
+			Expect(err).To(BeNil())
+			Expect(len(httpProxyList.Items)).To(Equal(1))
+			Expect(httpProxyList.Items[0].Spec.VirtualHost.Fqdn).To(Equal(RouteFooFQDN))
+
+			err = k8sClient.Delete(context.Background(), &objRouteFoo)
+			Expect(err).To(BeNil())
+
+			err = k8sClient.Delete(context.Background(), &objRouteBar)
+			Expect(err).To(BeNil())
+
+			httpProxyList = contourv1.HTTPProxyList{}
+			err = k8sClient.List(context.Background(), &httpProxyList, client.InNamespace(DefaultNamespace))
+			Expect(err).To(BeNil())
+			Expect(len(httpProxyList.Items)).To(Equal(1))
+		})
+
+		It("should create new HTTPProxy object when the router host is changed, and the older HTTPProxy should get deleted", func() {
+			objRouteFoo := routev1.Route{
+				TypeMeta: RouterTypeMeta,
+				ObjectMeta: v1.ObjectMeta{
+					Namespace: DefaultNamespace,
+					Name:      RouteFooName,
+					Labels: map[string]string{
+						consts.RouteShardLabel: RouterName,
+					},
+					Annotations: map[string]string{
+						consts.AnnotTimeout: RouteTimeout,
+					},
+				},
+				Spec: routev1.RouteSpec{
+					Host: RouteFooFQDN,
+					Port: &routev1.RoutePort{
+						TargetPort: intstr.IntOrString{IntVal: 443},
+					},
+					To: routev1.RouteTargetReference{
+						Name:   ServiceFooName,
+						Kind:   KindService,
+						Weight: &ServiceWeight,
+					},
+					WildcardPolicy: routev1.WildcardPolicyType(RouteFooWildCardPolicyType),
+				},
+			}
+			err = k8sClient.Create(context.Background(), &objRouteFoo)
+			Expect(err).To(BeNil())
+
+			objRouteFoo.Status = routev1.RouteStatus{
+				Ingress: []routev1.RouteIngress{
+					{
+						RouterName: RouterName,
+						Conditions: []routev1.RouteIngressCondition{
+							{
+								Status: v12.ConditionStatus(v1.ConditionTrue),
+								Type:   routev1.RouteAdmitted,
+							},
+						},
+					},
+				},
+			}
+			err = k8sClient.Status().Update(context.Background(), &objRouteFoo)
```

Comment:
I think there's no need to tweak the openshift router's behavior. Creating the route with its status field set should be enough. Agree?